contracts/ExitFormat.sol

```diff
@@ -97,34 +97,51 @@ library ExitFormat {
      * @param exit The exit to be paid out.
      */
     function executeExit(ExitFormat.SingleAssetExit[] memory exit) public {
-        for (uint256 i = 0; i < exit.length; i++) {
-            address asset = exit[i].asset;
-            for (uint256 j = 0; j < exit[i].allocations.length; j++) {
-                require(
-                    _isAddress(exit[i].allocations[j].destination),
-                    "Destination is not a zero-padded address"
-                );
-                address payable destination =
-                    payable(
-                        address(
-                            uint160(uint256(exit[i].allocations[j].destination))
+        for (uint256 assetIndex = 0; assetIndex < exit.length; assetIndex++) {
+            executeSingleAssetExit(exit[assetIndex]);
+        }
+    }
+
+    /**
+     * @notice Executes a single asset exit by paying out the asset and calling external contracts
+     * @dev Executes a single asset exit by paying out the asset and calling external contracts
+     * @param singleAssetExit The single asset exit to be paid out.
+     * TODO absorb into exit format repo
+     */
+    function executeSingleAssetExit(
+        ExitFormat.SingleAssetExit memory singleAssetExit
+    ) public {
+        address asset = singleAssetExit.asset;
+        for (uint256 j = 0; j < singleAssetExit.allocations.length; j++) {
+            require(
+                _isAddress(singleAssetExit.allocations[j].destination),
+                "Destination is not a zero-padded address"
+            );
+            address payable destination =
+                payable(
+                    address(
+                        uint160(
+                            uint256(singleAssetExit.allocations[j].destination)
                         )
+                    )
+                );
+            uint256 amount = singleAssetExit.allocations[j].amount;
+            if (asset == address(0)) {
+                (bool success, ) = destination.call{value: amount}(""); //solhint-disable-line avoid-low-level-calls
```

Comment:
We spoke about this briefly the other day but I will recap my (mis?)understanding:
- `destination.transfer(amount)` is considered unsafe because it forwards a (small) set amount of gas, which can cause a revert in cases where the referenced address is a contract. Especially troubling is that gas costs can change in the future.
- `destination.transfer(amount)` is safe in cases where address is known to be an external account (because it doesn't execute code which can run out of gas), but `.call{value:amount}` is safe in this case as well because EOAs aren't reentrency threats

This is a point I remain slightly fuzzy on:
- in nitro, destinations are either participants (EOAs) or other channels (full bytes32 values). The destination referenced in this line has been verified as an EOA. `destination.transfer(amount)` would be safe in this case.
- but, more generally, the exit-format aims for compatibility with protocols where this property of account destinations always referring to EOAs does not hold, which requires the `.call{value: amount}` pattern

Reply:
`destination.transfer` forwards a small amount of gas, as you say. That can be considered as a safety feature, because minimal gas means a minimal ability to commit reentrancy.

However, the auditors recommend not *relying* on that mechanism to avoid reentrancy. So we have our checks/effects/interaction pattern to protect us instead. 

So then it comes down to whether you want to rely on hardcoded gas amounts or not. We've chosen a method that doesn't rely on hardcoded gas amounts -- which seems better because gas costs can and do change. I think it is worth saying that the decision here seems very marginal to me, from a security point of view. There's not a **really** compelling reason to choose `destination.call{value:amount}("")` over `destination.transfer(amount)`.

Point of clarification about nitro destinations. I think you may be conflating what we call "external destinations" with EOAs (externally owned accounts, i.e. ones controlled by a private key). An external destination can still be a contract -- it could be a smart contract wallet, for example. None of our code tries to "switch" on destination addresses being contracts versus EOAs.